internal/configuration/flake.go

```diff
@@ -19,18 +20,46 @@ type FlakeRef struct {
 	Builder system.CommandRunner
 }
 
+func ResolvePath(path string) (string, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return path, nil
+	}
+
+	resolved, err := filepath.EvalSymlinks(path)
+
+	if err != nil {
+		return path, err
+	}
+
+	return resolved, nil
+}
+
 func FlakeRefFromString(s string) *FlakeRef {
 	split := strings.Index(s, "#")
 
+	var uri string
+	if split > -1 {
+		uri = s[:split]
+	} else {
+		uri = s
+	}
+
+	if !strings.Contains(uri, "://") {
+		resolved, err := ResolvePath(uri)
+		if err == nil {
+			uri = resolved
+		}
+	}
+
```

Comment:
IMO we can probably just attempt to call os.Stat(), and if it fails, just use the URI as is. It's highly unlikely that someone is going to have a path name with `://` in the name anyway.

Open to suggestions though!

```suggestion
	if _, err := os.Stat(path); err == nil {
		if resolved, err := filepath.EvalSymlinks(path); err == nil {
            uri = resolved
        }
	}
```